### PR TITLE
Added AvroCompatibilityHelper(https://github.com/linkedin/avro-util) support on Hive 1.0.1 to help hive consumers upgrade to higher Avro version

### PR DESCRIPTION
**Pull request by abhisheknath2011**
…support for hive to replace deprecated methods and help Avro version upgrade.

### What changes were proposed in this pull request?
1. Usage of AvroCompatibilityHelper to replace deprecated Avro methods.
2. AvroCompatibilityHelper to help hive consumer easily migrate to the higher version of Avro.
3. No need to upgrading from Jackson 1.x to Jackson 2.x as compatibility helper handles internally.
4. Changes are done one local fork of hive 1.0.1 branch. 

### Why are the changes needed?
1. Gobblin currently uses hive 1.0.1 and the upgrade of Avro 1.9.2 is blocked due older version of Avro and Jackson on hive 1.0.1.
 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
1. Local build
2. Released a local snapshot and applied on Gobblin.
3. With these changes gobblin build failures coming from hive serde library is resolved.

### CODE REVIEW

**Review by omalley — CHANGES_REQUESTED** [review]
You need to use branch-1.0 as the base, not branch-1.0.1, which is an artifact of the subversion to git migration.

**Review by omalley — CHANGES_REQUESTED** [review]
Please revert all of the white space changes.

You need a version of this patch for the master branch. It would be good to also have a port of this fix to at least the major branches between.
* branch-3
* branch-3.1
* branch-2
* branch-2.8

File: pom.xml [hunk 1]
```diff
@@ -166,6 +167,30 @@
   </properties>
 
   <repositories>
+    <repository>
```
[the remark this answers is not in the document]
Comment (abhisheknath2011) [reply pom.xml:170]: Added LinkedIn jfrog external repo.

File: accumulo-handler/pom.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
   <parent>
     <groupId>org.apache.hive</groupId>
     <artifactId>hive</artifactId>
-    <version>1.0.1</version>
+    <version>1.0.1-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (abhisheknath2011) [reply accumulo-handler/pom.xml]: Changed the release version to 1.0.2

File: pom.xml [hunk 3]
```diff
@@ -166,6 +167,19 @@
...
+      <id>avro-util-repository</id>
+      <name>datanucleus maven repository</name>
```
Comment (srramach) [inline pom.xml:172]: Update the name. This seems to be a copy-paste of the item below.

Comment (abhisheknath2011) [reply pom.xml]: Changed to Avro util repo name.

**Comment by srramach** [comment]
Not sure if it's necessary or correct, but look at the `EQUIVALENT_VERSIONS` map in `MetaStoreSchemaInfo.java` and see whether to include a `"1.0.2" -> "1.0.0"` entry there.

**Comment by abhisheknath2011** [comment]
> Not sure if it's necessary or correct, but look at the `EQUIVALENT_VERSIONS` map in `MetaStoreSchemaInfo.java` and see whether to include a `"1.0.2" -> "1.0.0"` entry there.

Thanks for checking this. Added equivalent schema version for this version (1.0.2 -> 1.0.0) to keep the consistency with the existing code and also this version did not change any schema.

**Pull request closed without merging**